Add functionality for postgresql database

tanweeralii opened the pull request:
Issue #13

tanweeralii committed:
Added functionality for postgresql database

hariprasanths commented on src/database/inquirer.js:
```diff
@@ -71,6 +71,7 @@ let askConfig = async (jobName, key) => {
             let defaultPort;
             if (ans.dbType == 'MongoDB') defaultPort = '27017';
             else if (ans.dbType == 'MySQL') defaultPort = '3306';
+            else if(ans.dpType == 'PostgreSQL') defaultPort = '5432';
```
Typo - `dptype`

hariprasanths commented on src/database/validator.js:
```diff
@@ -22,7 +22,7 @@ const validateInitConfig = async (config) => {
         throw new Error(`Invalid config: Missing required field - '${dbConfigKeys.dbType}'`);
     }
-    if (!['MongoDB', 'MySQL'].includes(config[dbConfigKeys.dbType])) {
+    if (!['MongoDB', 'MySQL', 'PostgreSQL'].includes(config[dbConfigKeys.dbType])) {
```
Update README under features to show that PostgreSQL is supported and examples to include PostgreSQL sample 
configuration.

tanweeralii committed:
Fixed typo

hariprasanths commented:
Could you please make the commit messages imperative?

Try to follow these guidelines when writing commit messages:
1. Separate subject from body with a blank line
2. Limit the subject line to 50 characters
3. Capitalize the subject line
4. Do not end the subject line with a period
5. Use the imperative mood in the subject line
6. Wrap the body at 72 characters
7. Use the body to explain what and why vs. how

Source: https://chris.beams.io/posts/git-commit/

tanweeralii commented:
Ok sure

hariprasanths commented on src/database/database.js:
```diff
@@ -2,6 +2,7 @@ const strings = require('./../utils/strings');
 const configstore = require('conf');
 const mongoDb = require('./mongoDb/mongoDb');
 const mysql = require('./mysql/mysql');
+const postgresql = require('./postgresql/postgrsql');
```
Typo

hariprasanths replied:
Is this resolved?

hariprasanths replied:
Do test the application before committing.

tanweeralii replied:
ohhh I am doing it right now

hariprasanths commented on src/database/postgresql/postgresql.js:
```diff
@@ -0,0 +1,108 @@
+};
+
+let dump = async (dbConfig, key, backupPath) => {
+    const postgresqlDumpCmd = `PGPASSWORD=${dbConfig.dbAuthPwd} \
```
Make this command compatible for powershell(windows) as well.

hariprasanths commented on src/database/postgresql/postgresql.js:
```diff
@@ -0,0 +1,108 @@
+        await files.decompressFile(backupFilePath);
+        backupFilePath = `${backupFilePath}.sql`;
+    }
+    const postgresqlRestoreCmd = `PGPASSWORD=${dbConfig.dbAuthPwd} \
```
Make this command compatible for powershell(windows) as well.

tanweeralii replied:
ok

tanweeralii committed:
Document PostgreSQL database support

Add `PostgreSQL` in "Support Database" part of `README.md`
Add Example to configure PostgreSQL in `docs/examples.md`
Add `PostgreSQL` in "Database Configuratin" area of `docs/configuration-using-file.md`

tanweeralii committed:
Update `postgresqlDumpCmd` and `postgresqlRestoreCmd` in "/src/database/postgresql/postgresql.js"

Modify `postgresqlDumpCmd` and `postgresqlRestoreCmd` to make compatible with Powershell

tanweeralii committed:
Fix typo in /src/database/database.js

Change `postgrsql` to `postgresql`

hariprasanths commented on src/database/postgresql/postgresql.js:
```diff
@@ -0,0 +1,99 @@
+let dump = async (dbConfig, key, backupPath) => {
+    const postgresqlDumpCmd = `pg_dump --dbname=postgresql://${dbConfig.dbAuthUser}:${dbConfig.dbAuthPwd}@${dbConfig.dbHost}:${dbConfig.dbPort}/${dbConfig.dbName} \
+    --compress=0..9 \
+    --format=c
```
Escape the new line

hariprasanths replied:
Always test the features before committing.

tanweeralii replied:
Yeah I did. I have checked this file and is not showing me this error.

tanweeralii replied:
Ok. I fixed it but I will commit after testing the feature

hariprasanths replied:
Okay, the backup command doesnt work for me (in bash shell) without the escaping the new line.
Try running the synchly commands to test out the features.
For example: Backup a db using the `synchly --run` command to take an instant backup and restore it using `synchly -R`.

tanweeralii replied:
ok

tanweeralii replied:
Hey, I tried doing backup and restore. I am getting error in restore but backup is successful.

tanweeralii replied:
for every table present in database it is showing that the table already exists.

tanweeralii replied:
![Screenshot from 2020-11-22 15-52-55](https://user-images.githubusercontent.com/46401579/99901201-def4a600-2cda-11eb-8290-4c460d3140e7.png)

hariprasanths commented:
Ohh, try dropping the db before restoring.
I remember seeing support for the same in pg_restore command.

On Sun, 22 Nov, 2020, 3:54 PM TANWEER ALI, <notifications@github.com> wrote:

> *@tanweeralii* commented on this pull request.
> ------------------------------
>
> In src/database/postgresql/postgresql.js
> <https://github.com/hariprasanths/synchly/pull/22#discussion_r528313944>:
>
> > +        password: dbConfig.dbAuthPwd,
> +        host: dbConfig.dbHost,
> +        port: dbConfig.dbPort,
> +        database: dbConfig.dbName,
> +    });
> +
> +    const connRes = await awaitPostgresqlConnect(connection);
> +    const disConnRes = await awaitPostgresqlEnd(connection);
> +
> +    return connRes;
> +};
> +
> +let dump = async (dbConfig, key, backupPath) => {
> +    const postgresqlDumpCmd = `pg_dump --dbname=postgresql://${dbConfig.dbAuthUser}:${dbConfig.dbAuthPwd}@${dbConfig.dbHost}:${dbConfig.dbPort}/${dbConfig.dbName} \
> +    --compress=0..9 \
> +    --format=c
>
> [image: Screenshot from 2020-11-22 15-52-55]
> <https://user-images.githubusercontent.com/46401579/99901201-def4a600-2cda-11eb-8290-4c460d3140e7.png>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hariprasanths/synchly/pull/22#discussion_r528313944>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGYRIDRHQKI3X6CZUSI7FUDSRDRGFANCNFSM4TWNDULA>
> .
>

tanweeralii committed:
Fix escape new line error in `dump` function of "src/postgresql/postgresql.js" by adding `\`
Add `--clean` in `restore` function of "src/postgresql/postgresql.js" to drop tables before restoring

tanweeralii commented:
Now both dump and restore functions are working properly.

hariprasanths commented on package.json:
```diff
@@ -60,17 +60,18 @@
-        "omelette": "^0.4.12",
-        "ora": "^4.0.4",
-        "ssh2-sftp-client": "^5.1.2"
+        "node-mailer": "^0.1.1",
```
Can you check this dependency. It does not look like the package we are using for nodemailer.

tanweeralii replied:
sure.

tanweeralii committed:
Update `package.json` file

Remove "node-mailer": "^0.1.1" line

hariprasanths commented on package.json:
```diff
@@ -60,18 +60,17 @@
-        "nodemailer": "^6.4.16",
-        "omelette": "^0.4.15",
-        "ora": "^4.1.1",
+        "node-mailer": "^6.4.6",
```
Kindly verify this package.
Seems like [this](https://www.npmjs.com/package/nodemailer) is the official package.

tanweeralii replied:
ok sure

tanweeralii replied:
done

tanweeralii committed:
Change "node-mailer" package to "nodemailer" in package.json

hariprasanths commented on package.json:
```diff
@@ -66,7 +66,7 @@
         "mysql": "^2.18.1",
         "node-cron": "^2.0.3",
-        "node-mailer": "^6.4.6",
+        "nodemailer": "^6.4.6",
```
Can you please check the version of nodemailer. Seems like 6.4.17 is the latest version. [Nodemailer documentation](https://www.npmjs.com/package/nodemailer)

tanweeralii replied:
sure

tanweeralii committed:
Revert

tanweeralii committed:
Revert "Revert"

This reverts commit 3ce086e5a26f4ff5710af5a4a8afcd985082980b.

tanweeralii committed:
Revert "Change "node-mailer" package to "nodemailer" in package.json"

This reverts commit 28bef94596c6e9fadedd458819b0fe5a45b3c7e6.

tanweeralii committed:
Update nodemailer version

tanweeralii commented:
I think postgresql part is completed. Now i will start working on s3 feature.

hariprasanths commented:
Can you please remove the package-lock.json file from the changes.

tanweeralii committed:
Add `package-lock.json` in `.gitignore`

tanweeralii committed:
Remove `package-lock.json` file
